plugin/view_config/view_config.py

```diff
@@ -383,11 +383,15 @@ def __get_default_flags(view, settings, need_lang_flags):
         # If the user provided explicit target compilers, retrieve their
         # default flags and append them to the list:
         target_compiler = settings.target_compilers[lang_tag]
+        flags = []
+        if isinstance(target_compiler, dict):
+            compiler_flags = target_compiler["flags"]
+            target_compiler = target_compiler["compiler"]
+
+
         if target_compiler is None and settings.use_default_includes:
             target_compiler = settings.clang_binary
 
-        compiler_flags = settings.compiler_args_to_get_builtins[lang_tag]
```

Comment:
We should be careful about this as it seems to me that what you propose now subtly changes the behavior. We would need to indicate at least in the docs that specifying the flags in the compiler settings explicitly will disable getting builtin flags from the compiler. We might even want to assert, that the setting to use builtins is actually false and indicate this to the user to avoid any confusion.